features/home/providers/vestings-provider.tsx

```diff
@@ -19,47 +20,73 @@ export const iconsMap = {
 };
 
 export type VestingsValue = {
-  currentVesting?: string;
-  vestings?: string[];
   setCurrentVesting: (vesting: string) => void;
-  isClaiming: boolean;
-  setIsClaiming: (value: boolean) => void;
-  isVestingsLoading: boolean;
-};
+} & (
+  | {
+      isVestingsLoading: false;
+      currentVesting: string;
+      vestings: string[];
+    }
+  | {
+      isVestingsLoading: false;
+      currentVesting: undefined;
+      vestings: [];
+    }
+  | {
+      isVestingsLoading: true;
+      currentVesting: undefined;
+      vestings: undefined;
+    }
+);
 
-const VestingsProvider: FC<PropsWithChildren> = ({ children }) => {
+export const VestingsProvider: FC<PropsWithChildren> = ({ children }) => {
   const [currentVesting, setCurrentVesting] = useState<string>();
   const [vestings, setVestings] = useState<string[]>();
-  const [isClaiming, setIsClaiming] = useState(false);
+  const [isVestingsLoading, setIsVestingsLoading] = useState<boolean>(true);
+  const { active } = useWeb3();
 
-  const { data, isValidating } = useVestings();
+  const { data, isLoading, isValidating, error } = useVestings();
 
   useEffect(() => {
-    if (data && data.length) {
-      setCurrentVesting(data[0].escrow);
+    if (isLoading) {
+      setIsVestingsLoading(true);
+      setCurrentVesting(undefined);
+      setVestings(undefined);
+      return;
+    }
 
-      const vestings = data.map((vesting) => vesting.escrow);
-      setVestings(vestings);
+    if (!active || !data?.length) {
+      setIsVestingsLoading(false);
+      setCurrentVesting(undefined);
+      setVestings([]);
+      return;
     }
-  }, [data]);
+    setIsVestingsLoading(false);
+    setCurrentVesting(data[0].escrow);
+    setVestings(data.map((vesting) => vesting.escrow));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [
+    active,
+    isLoading,
+    isValidating,
+    error,
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    data?.map((vesting) => vesting.escrow).join(';'),
+  ]);
 
-  const value = useMemo(
-    () => ({
-      currentVesting,
-      vestings,
-      setCurrentVesting,
-      isClaiming,
-      setIsClaiming,
-      isVestingsLoading: isValidating,
-    }),
-    [currentVesting, isClaiming, isValidating, vestings],
-  );
+  const value = {
+    isVestingsLoading,
+    currentVesting,
+    setCurrentVesting,
+    vestings,
+  };
 
   return (
-    <VestingsContext.Provider value={value}>
+    <VestingsContext.Provider value={value as VestingsValue}>
```

Comment:
`value={value}`
Or do you have a problem with types?

Reply:
`VestingsValue` is more concrete, because there are just 3 states, but since this value is built upon 3 `useState` hooks, from the typescript perspective this typing is incorrect, so I need to force it. There are ways to bypass it, but it feels like an overkill.